### ISSUE TITLE
Sub-Tasks not linked to Tasks

### ISSUE THREAD

**Issue by jstras17** [solidify/jira-azuredevops-migrator#745]
**Describe the problem**
Sub-Tasks not linked to Tasks
How do we link sub-tasks to Tasks if <issuelinks> are not present in the JIRA Task?

**To Reproduce**
When migrating Tasks and Sub-Tasks to ADO Sub-Tasks are not linked to Tasks
In the wi-import-log we see the following " link type () does not exist in project"

=================================
_[D][11:30:56]    Mapped AreaPath 'Sandbox\mig'.
[D][11:30:56]    Mapped IterationPath 'Sandbox\mig'.
[E][11:30:56] '[Added] GPPDSS-15/-1->GPPDSS-4/96425 []' - link type () does not exist in project
[W][11:30:56] ''GPPDSS-15', rev 0' - not all changes were saved.
[D][11:30:56]    Correcting comments on ''GPPDSS-15', rev 0'.
[D][11:30:57]    Imported revision._
=================================

When we look at the XML of the JIRA Task it shows the Subtasks
However, we don't see a <issuelinks> section like we do in other JIRA Issue Types
Instead there is a <subtasks> section (see below)

<subtasks>
<subtask id="2825329">GPPDSS-5</subtask>
<subtask id="3020667">GPPDSS-6</subtask>
<subtask id="3021316">GPPDSS-7</subtask>
<subtask id="3021749">GPPDSS-8</subtask>
<subtask id="3021949">GPPDSS-9</subtask>
<subtask id="3022280">GPPDSS-10</subtask>
<subtask id="3043074">GPPDSS-11</subtask>
<subtask id="3044612">GPPDSS-12</subtask>
<subtask id="3045347">GPPDSS-13</subtask>
<subtask id="3046421">GPPDSS-14</subtask>
<subtask id="3048681">GPPDSS-15</subtask>
</subtasks>

**Tool version**
 - Version of the jira-azuredevops-migrator tool 3.0.115

**Attachments**

- migration-json.txt
- jira-export-log.txt
- wi-import-log.txt

**Screenshots**
[migration-json.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11123599/migration-json.txt)
[wi-import-log-230331-113036.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11123632/wi-import-log-230331-113036.txt)
[jira-export-log-230331-113021.txt](https://github.com/solidify/jira-azuredevops-migrator/files/11123634/jira-export-log-230331-113021.txt)

**Comment by Alexander-Hjelm** [comment]
@jstras17 

Is this issue related to https://github.com/solidify/jira-azuredevops-migrator/issues/495?

Otherwise, try these issues:
- https://github.com/solidify/jira-azuredevops-migrator/issues/170
- https://github.com/solidify/jira-azuredevops-migrator/issues/166

**Comment by jstras17** [comment]
Any updates on this Support Issue?

Thanks

**Comment by jstras17** [comment]
Not creating work item links manually post-migration

Asking if it's possible to configure the JIRA Migration Tool to link Sub-Tasks to Tasks during the migration?

**Comment by Alexander-Hjelm** [comment]
Yep, that is supported and works as expected in our integration tests.

Could possibly be related to https://github.com/solidify/jira-azuredevops-migrator/issues/495

What project template are you using? (Scrum/Agile/CMMI/Basic...)

**Comment by jstras17** [comment]
Using Scrum Template

See attached migration.json
Included are the .json files for 2 Issues (1 Task and 1 Sub-Task)
And XML for Task and Sub-Task

Sub-Tasks are being created as Tasks in ADO (as mapped in migration.json)
However, Sub-Task is not linking to to Task

=============================

Seeing error in import log:
[E][10:07:19] '[Added] GPPDSS-5/-1->GPPDSS-4/97157 []' - link type () does not exist in project

If you look at the XML for JIRA Task and Sub-Task there is no link type section
There is however, a line for "parent id"
<parent id="2757466">GPPDSS-4</parent>

And another section for <subtasks>
<subtasks>
<subtask id="2825329">GPPDSS-5</subtask>
<subtask id="3020667">GPPDSS-6</subtask>
<subtask id="3021316">GPPDSS-7</subtask>
<subtask id="3021749">GPPDSS-8</subtask>
<subtask id="3021949">GPPDSS-9</subtask>
<subtask id="3022280">GPPDSS-10</subtask>
<subtask id="3043074">GPPDSS-11</subtask>
<subtask id="3044612">GPPDSS-12</subtask>
<subtask id="3045347">GPPDSS-13</subtask>
<subtask id="3046421">GPPDSS-14</subtask>
<subtask id="3048681">GPPDSS-15</subtask>
</subtasks>


=============================

Is the problem that there are no linktypes in the JIRA Issues?


Attached

- migration.json
- .xml for Task and Sub-Task
- export log
- import log
[case745.zip](https://github.com/solidify/jira-azuredevops-migrator/files/11274172/case745.zip)

**Comment by Alexander-Hjelm** [comment]
We should probably add a SCRUM project to our integration tests.

> Is the problem that there are no linktypes in the JIRA Issues?

I'm not sure about this. How did you obtain these XML files?

**Comment by Alexander-Hjelm** [comment]
Ok! This does not seem to be an issue with the jira-ado migrator tool itself then, if I understand correctly? But rather with your process configuration?

**Comment by jstras17** [comment]
JIRA Sub-tasks are not showing up as links to Tasks

Is that because the Sub-tasks do no have "issuelinks" in the sub-task XML?
<issuelinks>

Have noticed that the Sub-task does have "parent id" in the sub-task XML
<parent id="2757466">GPPDSS-4</parent>

Is there a way to configure jira-ado migrator to use this "parent id" element to link Sub-task to Tasks?
Is it possible we do not have jira-ado migrator configured correctly?

**Comment by jstras17** [comment]
JIRA Sub-tasks are not showing up as links to Tasks

Is that because the Sub-tasks do no have "issuelinks" in the sub-task XML?  *See image example

Have noticed that the Sub-task does have "parent id" in the sub-task XML   *See image example
*See image example

Is there a way to configure jira-ado migrator to use this "parent id" element to link Sub-task to Tasks?
Is it possible we do not have jira-ado migrator configured correctly?

![image](https://user-images.githubusercontent.com/68659481/233688705-2f95b7d0-21b9-4c93-97aa-41c9f296d22b.png)
![image](https://user-images.githubusercontent.com/68659481/233688778-7c99db15-fca2-4c11-8984-58bf4dbc1082.png)

**Comment by Alexander-Hjelm** [comment]
That could be the problem indeed!

I assume you use the "Parent Link" field to manage task-subtask relationships? And that you have no actual "issue links" set up?
